bach/bach/dataframe.py

```diff
@@ -1114,6 +1114,52 @@ def get_unsampled(self) -> 'DataFrame':
         from bach.sample import get_unsampled
         return get_unsampled(df=self)
 
+    def database_create_table(self, table_name: str, overwrite: bool = False, ) -> 'DataFrame':
+        """
+        Write the current state of the DataFrame to a database table.
+
+        The DataFrame that's returned will query from the written table for any further operations.
+
+        :param table_name: Name of the table to write to. Can include project_id and dataset
+                on BigQuery, e.g. 'project_id.dataset.table_name'
+        :param overwrite: if True, the data is written to `table_name`, even if that table already
+            exists. Beware, this will drop an existing table if `table_name` already exists. Make sure that
+            `table_name` does not contain any valuable information. Additionally, make sure that it is not a
+            source table of this DataFrame.
+        :return: New DataFrame; the base_node consists of a query on the newly created table.
+
+        .. warning::
+            With `overwrite=True`, if a table already exist with the given name, then that table will
+            be dropped and all data lost!
+        .. note::
+            This function queries the database.
+        .. note::
+            This function writes to the database.
+        """
+        dialect = self.engine.dialect
+        model = self.get_current_node(name='database_create_table')
+        model = model.copy_set_materialization(Materialization.TABLE)
+        model = model.copy_set_materialization_name(materialization_name=table_name)
+
+        placeholder_values = get_variable_values_sql(dialect=dialect, variable_values=self.variables)
+        model = update_placeholders_in_graph(start_node=model, placeholder_values=placeholder_values)
+
+        sql = to_sql(dialect=dialect, model=model)
+        with self.engine.connect() as conn:
+            if overwrite:
+                sql = f'DROP TABLE IF EXISTS {quote_identifier(dialect, table_name)}; {sql}'
+
+            sql = escape_parameter_characters(conn, sql)
+            conn.execute(sql)
+
+        all_dtypes = {**self.index_dtypes, **self.dtypes}
+        return self.from_table(
```

Comment:
IMHO, looks a bit odd calling a `classmethod` from `self`
```suggestion
        return DataFrame.from_table(
```

Reply:
I think it is actually more correct to use `self.from_table(` in cases like this. If for some reason somebody decides to subclass DataFrame and override `from_table`, then with `self.` we'll call the function in the subclass. Which seems like more logical behaviour to me

Reply:
I think in some places we do follow that logic, but not here. For example in Series we almost always call classmethods on the specific class or on self, never `Series.<classmethod>()`. But with DataFrame we most definitely don't do that consistently, and it might actually quite a bit of refactoring to do that consistently.